### PR TITLE
F/cicd rebuild

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

Updates the CI/CD workflows to only release the contracts if one of these labels are set:
- release:patch
- release:minor
- release:major

Task: DOPS-303

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: .github/workflows/npm-release-reusable.yml [hunk 1]
```diff
@@ -62,7 +62,9 @@ jobs:
         run: |
           yarn config set version-git-tag false
           # run the bump twice to remove -alpha first
-          yarn version --${{ needs.prepare.outputs.bump }}
+          if [[ $(cat package.json | jq -r .version) == *"-alpha" ]]; then
+            yarn version --${{ needs.prepare.outputs.bump }}
+          fi
```
Comment (brickpop) [inline .github/workflows/npm-release-reusable.yml:67]: Does this version bump affect the contracts only? Or also the bindings for the Ethers NPM package?

Comment (mathewmeconry) [reply .github/workflows/npm-release-reusable.yml:67]: the packages `@aragon/core-contracts` and `@aragon/core-contracts-ethers`

File: .github/workflows/contracts-deploy.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ on:
 
 jobs:
   test:
-    if: github.actor != 'arabot-1' && github.event.pull_request.merged
+    if: github.actor != 'arabot-1' && github.event.pull_request.merged && contains(github.event.pull_request.labels.*.name, 'release:')
```
[the remark this answers is not in the document]
Comment (mathewmeconry) [reply .github/workflows/contracts-deploy.yml:15]: it skips the whole deployment and testing thing after merge
but it doesn't stop testing on a different branch than `develop` or  `main`

[the remark this answers is not in the document]
Comment (brickpop) [reply .github/workflows/contracts-deploy.yml:15]: I would still run the tests on `main` and `develop`. The filters would apply to deployment/release only ;-)

[the remark this answers is not in the document]
Comment (mathewmeconry) [reply .github/workflows/contracts-deploy.yml:15]: done. changed the workflow `contract-tests.yml` to always run

**Review by brickpop — APPROVED** [review]

**Pull request merged**